Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/UnityProjectInfo.cs

```diff
@@ -164,7 +162,26 @@ public void RefreshProjects()
                             throw new InvalidOperationException($"Failed to retrieve AsmDef for script assembly: {pair.Key}");
                         }
 
-                        assemblyDefinitionInfo = AssemblyDefinitionInfo.GetDefaultAssemblyCSharpInfo(pair.Value);
+                        Guid guid;
+                        switch (pair.Key)
+                        {
+                            case "Assembly-CSharp":
+                                guid = config.AssemblyCSharpGuid;
+                                break;
+                            case "Assembly-CSharp-firstpass":
+                                guid = config.AssemblyCSharpFirstPassGuid;
+                                break;
+                            case "Assembly-CSharp-Editor":
+                                guid = config.AssemblyCSharpEditorGuid;
+                                break;
+                            case "Assembly-CSharp-Editor-firstpass":
+                                guid = config.AssemblyCSharpFirstPassEditorGuid;
+                                break;
+                            default:
+                                throw new InvalidOperationException($"Predefined assembly '{assemblyDefinitionInfo.Name}' was not recognized, this generally means it should be added to the switch statement in CSProjectInfo:GetProjectType. Treating is as a PredefinedAssembly instead of PredefinedEditorAssembly.");
```

Comment:
>is [](start = 248, length = 2)

I think it is supposed to be "Treating it as a PredefinedAssembly" not "is"

Reply:
You are right, however, the message is actually stale now, since I just throw. I will update the message, thanks for flagging it.